cantor-base/src/main/java/com/salesforce/cantor/Objects.java

```diff
@@ -118,6 +118,20 @@ default void delete(String namespace, Collection<String> keys) throws IOExceptio
      */
     Collection<String> keys(String namespace, int start, int count) throws IOException;
 
+    /**
+     * Returns paginated list of entries; the returned list is not ordered.
+     *
+     * @param namespace the namespace identifier
+     * @param prefix to narrow down the search results
+     * @param start start offset
+     * @param count maximum number of entries to return; -1 for infinite
+     * @return paginated list of keys in the namespace
+     * @throws IOException exception thrown from the underlying storage implementation
+     */
+    default Collection<String> keys(String namespace, String prefix, int start, int count) throws IOException {
+        return keys(namespace, start, count);
```

Comment:
I would prefer this logic to be flipped so the keys call without prefix defaults to calling the one with the prefix parameter, but that would require implementations for every database type. This was much simpler and allows us to slowly convert each type and vet them independently. Any thoughts?